src/expire.c

```diff
@@ -321,6 +336,23 @@ void activeExpireCycle(int type) {
              * not reclaimed). */
         } while (data.sampled == 0 ||
                  (data.expired * 100 / data.sampled) > config_cycle_acceptable_stale);
+    
+        /* Update the average TTL stats for this database. 
+         * If we have reached the time limit, we should return ASAP.*/
```

Comment:
i'd like to avoid cloning this code.
i think it's possible if we change the `while (data.sampled == 0 ||` loop, in favor of an explicit exit condition, which will decide if it wants to exit, before computing the average (and do the exit after)

Reply:
Sorry, my mind was a little confused when I wrote this code. I have fixed it and removed the redundant cloning code.